src/products/productmanagement.js

```diff
@@ -0,0 +1,136 @@
+import { retreiveCachedItems } from "../core/services.js";
+import { menus } from "../menus/menuoptions.js";
+import { con } from "../menus/rendermenu.js";
+
+const optionProducts = [];
+export const offersArray = [];
+export const cartItems = [];
+export const totalCost = {};
+export const itemSelection = {};
+
+async function fetchCategories() {
+  let results = "";
+  try {
+    const response = await axios.get(`${BASEURL}/ussd/prodcategories`);
+    response.data.data.data.forEach((category) => {
+      optionProducts.push(category.id);
+      results += `\n${category.id}. ${category.category_name}`;
+    });
+    return results;
+  } catch (error) {
+    return error;
+  }
+}
+async function fetchProducts(id) {
+  let results = "";
+  try {
+    const response = await axios.get(`${BASEURL}/ussd/prodcategories`);
+
+    response.data.data.data.forEach((item) => {
+      item.products.forEach((description) => {
+        if (description.category_id === id) {
+          results += `${description.id}. ${description.product_name}\n `;
+        }
+      });
+    });
+    return results;
+  } catch (error) {
+    throw new Error(error);
+  }
+}
+const fetchFarmOfferings = async (id) => {
+  let farmOfferings = "";
+  try {
+    const response = await axios.get(`${BASEURL}/ussd/prodcategories`);
+    response.data.forEach((item) => {
+      item.farm_products.forEach((farmItem) => {
+        if (farmItem.product_id === id) {
+          farmOfferings += `${farmItem.farm_id}. Available units: ${farmItem.units} grade of items: ${farmItem.grade}\n `;
+        }
+      });
+    });
+    return farmOfferings;
+  } catch (error) {
+    throw new Error(error);
+  }
+};
+
+const addProduct = async (productdata) => {
+  try {
+    const newProduct = axios.post(
+      `${BASEURL}/ussd/farmcatalog/save`,
+      productdata
+    );
+
+    return newProduct;
+  } catch (err) {
+    throw new Error(err);
+  }
+};
+
+const getSpecificProduct = async (id) => {
+  try {
+    const specificProduct = await axios.get(`${BASEURL}/ussd/products/all`);
+    const filteredItems = specificProduct.data.filter((item) => item.id === id);
+    let respose = "";
+    filteredItems.forEach((filteredItem) => {
+      respose += `${filteredItem.id}. ${filteredItem.product_name}`;
+    });
+    return respose;
+  } catch (err) {
+    return false;
+  }
+};
+
+export const confirmQuantityWithPrice = async (
+  userQuantity,
+  productID,
+  status,
+  client
+) => {
+  let availableUnits = 0;
+  let pricePoint;
+  let message;
+  let offers = await retreiveCachedItems(client, ["offersArray"]);
+  offers = JSON.parse(offers);
+  const buyerSelection = offers.filter((item) => item.id === productID);
+
+  availableUnits = buyerSelection[0].availableUnits;
+
+  if (userQuantity > availableUnits) {
+    message = `${con()} The amount you set is higher than the available units go back and choose a smaller quantity`;
+  } else {
+    if (status === "unit") {
```

Comment:
Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/JoeMindo/USSDTest/pullRequest?prid=8695127)
```suggestion
    if (status === 'unit') {
```